src/components/password/password.component.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useRef, useState } from "react";
+import { TextboxProps } from "../textbox";
+import { StyledPassword, HiddenAriaLive } from "./password.style";
+import guid from "../../__internal__/utils/helpers/guid";
+import useLocale from "../../hooks/__internal__/useLocale";
+import ButtonMinor from "../button-minor/button-minor.component";
+
+export type TypeOptions = "text" | "password";
+
+export interface PasswordProps extends TextboxProps {
+  /** HTML type attribute of the input */
+  type?: TypeOptions;
+}
+
+export const Password = ({
+  id,
+  type = "password",
+  disabled,
+  ...rest
+}: PasswordProps) => {
+  const internalInputId = useRef(id || guid());
+  const l = useLocale();
+
+  const [passwordShown, setPasswordShown] = useState(type);
+  return (
+    <>
+      <StyledPassword
+        data-element="styled-password-container"
```

Comment:
Might be a good idea to use the `tagComponent` util here, so consumers can set the `data-element` and `data-role` props if they choose to as part of their tests
 
```suggestion
        {...tagComponent("password", rest)}
```